authentication/oauth.go

```diff
@@ -6,17 +6,31 @@ import (
 	"net/url"
 
 	"github.com/auth0/go-auth0/authentication/oauth"
+	"github.com/auth0/go-auth0/internal/idtokenvalidator"
 )
 
 // OAuth exposes logging in using OAuth based APIs.
 type OAuth manager
 
 // LoginWithGrant allows logging in with an OAuth 2.0 grant. This should only be needed if a grant
 // type is not supported byt this SDK.
-func (o *OAuth) LoginWithGrant(ctx context.Context, grantType string, body url.Values, opts ...RequestOption) (t *oauth.TokenSet, err error) {
+func (o *OAuth) LoginWithGrant(ctx context.Context, grantType string, body url.Values, optionalVerification oauth.IDTokenValidationOptionalVerification, opts ...RequestOption) (t *oauth.TokenSet, err error) {
```

Comment:
Maybe we can make `oauth.IDTokenValidationOptionalVerification` a little less verbose, like `oauth.IDTokenValidationOptions`?

Reply:
Yeah I didn't really like it, that's better